mk/config.mk

```diff
@@ -789,3 +789,11 @@ $(eval $(call cfg-enable-all-depends,CFG_MEMPOOL_REPORT_LAST_OFFSET, \
 CFG_TA_PAUTH ?= n
 
 $(eval $(call cfg-depends-all,CFG_TA_PAUTH,CFG_ARM64_core))
+
+# Enable support for generic watchdog registration
+# This watchdog will then be usable by non-secure world through SMC calls.
+CFG_WDT ?= n
+CFG_WDT_SM_HANDLER ?= n
+CFG_WDT_SM_HANDLER_ID ?= 0x82003D06
```

Comment:
The id seems to be specific to mt8173. It's likely going to be platform specific.

Reply:
It is quite ugly to be honest. For mt8173 it's the next free SIP function id. Who knows what's suitable for another platform?

Reply:
Agreed, I have switched the config option to the following which forces the user to define it:

```
ifeq ($(CFG_WDT_SM_HANDLER_ID),)
$(error CFG_WDT_SM_HANDLER_ID must be defined when enabling CFG_WDT_SM_HANDLER)
endif
```